build_runner_core/lib/src/asset_graph/graph.dart

```diff
@@ -292,20 +295,24 @@ class AssetGraph {
       deletedOutput.state = NodeState.definitelyNeedsUpdate;
     }
 
-    var allNewAndDeletedIds =
-        _addOutputsForSources(buildPhases, newIds, rootPackage)
-          ..addAll(transitiveRemovedIds);
-
     // Transitively invalidates all assets. This needs to happen after the
     // structure of the graph has been updated.
     var invalidatedIds = Set<AssetId>();
+
+    var newGeneratedOutputs =
+        _addOutputsForSources(buildPhases, newIds, rootPackage);
+    var allNewAndDeletedIds =
+        Set.of(newGeneratedOutputs.followedBy(transitiveRemovedIds));
+
     void invalidateNodeAndDeps(AssetId id) {
       var node = get(id);
       if (node == null) return;
       if (!invalidatedIds.add(id)) return;
 
-      if (node is NodeWithInputs && node.state == NodeState.upToDate) {
-        node.state = NodeState.mayNeedUpdate;
+      if (node is NodeWithInputs) {
+        if (node.state == NodeState.upToDate) {
```

Comment:
how is this different?

Reply:
Oh I originally had another condition in here but I removed it, reverting it back